sled-agent/src/http_entrypoints.rs

```diff
@@ -991,6 +991,20 @@ impl SledAgentApi for SledAgentImpl {
         ))
     }
 
+    async fn support_nvmeadm_info(
+        request_context: RequestContext<Self::Context>,
+    ) -> Result<HttpResponseOk<Vec<SledDiagnosticsQueryOutput>>, HttpError>
+    {
+        let sa = request_context.context();
+        Ok(HttpResponseOk(
+            sa.support_nvmeadm_info()
+                .await
+                .into_iter()
```

Comment:
If these end up returning a single `SledDiagnosticsQueryOutput` you can just return a single result like:

```rust
    async fn support_zoneadm_info(
        request_context: RequestContext<Self::Context>,
    ) -> Result<HttpResponseOk<SledDiagnosticsQueryOutput>, HttpError> {
        let sa = request_context.context();
        let res = sa.support_zoneadm_info().await;
        Ok(HttpResponseOk(res.get_output()))
    }
```